From: Ben Harris <bjh21@cam.ac.uk>
Date: Fri, 5 Apr 2019 20:32:41 +0000
Subject: improve detection of old style core dumps

Bug-Debian: bugs.debian.org/611292

The proposed patch is wrong: as written it arranges to reject filenames 
that begin with ".core", rather than those that end with ".core".  Files 
whose names end ".core" are already rejected by cfgfile_filter(), and 
files whose names begin ".core" are rejected by the general rejection of 
filenames beginning with ".".

I think the original test is intended to trap old-style core dumps, which 
are always called simply "core". This could be correctly achieved by:
Bug-Debian: bugs.debian.org/611292
---
 insserv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/insserv.c b/insserv.c
index 849b890..b47e080 100644
--- a/insserv.c
+++ b/insserv.c
@@ -3078,7 +3078,7 @@ int main (int argc, char *argv[])
 	    continue;
 	}
 
-	if (!strncmp(d->d_name, "core", strlen("core"))) {
+	if (!strcmp(d->d_name, "core")) {
 	    if (isarg)
 		warn("script name %s is not valid, skipped!\n", d->d_name);
 	    continue;
